janitor/utils.py

```diff
@@ -1680,3 +1687,176 @@ def _computations_as_categorical(df: pd.DataFrame, **kwargs) -> pd.DataFrame:
     df = df.astype(categories_dtypes)
 
     return df
+
+
+@functools.singledispatch
+def _select_columns(columns_to_select: str, df):
+    """
+    Base function for column selection.
+    Applies only to strings.
+    It is also applicable to shell-like glob strings,
+    specifically, the `*`.
+    A list/pd.Index of column names is returned.
+    """
+    filtered_columns = None
+    if "*" in columns_to_select:  # shell-style glob string (e.g., `*_thing_*`)
+        filtered_columns = fnmatch.filter(df.columns, columns_to_select)
+    elif columns_to_select in df.columns:
+        filtered_columns = [columns_to_select]
+    if not filtered_columns:
+        raise NameError(f"No match was returned for '{columns_to_select}'")
+    return filtered_columns
+
+
+@_select_columns.register(slice)
+def _(columns_to_select, df):
```

Comment:
Using `_` as the function name collides with the Python convention of using `_` as a variable to ignore. Could you change it to something like `def col_sel_dispatch`, or a more informative name?

I did a bit of googling myself, it looks like you might have read [the blog post on pythonlibrary.org](https://www.blog.pythonlibrary.org/2016/02/23/python-3-function-overloading-with-singledispatch/), is that right? :joy_cat:

A possibly saner convention with functions is described in [Michael Chow's blog post](https://mchow.com/posts/2020-02-24-single-dispatch-data-science/), and is what I'm thinking of above.

Reply:
Actually noticed it in the python [docs](https://docs.python.org/3/library/functools.html#functools.singledispatch) :grinning: